### PR TITLE
Plans: Add Happychat to sites plans page (take 2)

### PR DESCRIPTION
**Pull request by sirreal**
Reverts Automattic/wp-calypso#23537 (revert of previous PR #22988).

This is a second attempt at Automattic/wp-calypso#22988

Happiness announcement and coordination: pbg9X-cDn-p2

✅ ~Blocked by #23540~. Thanks Canary tests for detecting issues 🎉

### CODE REVIEW

**Comment by sirreal** [comment]
#23538 Has been merged, I'll plan to merge this next week.

**Comment by sirreal** [comment]
This is just another attempt at #22988 now that #23540 should fix issues discovered by Canary tests.

I'd appreciate a ✅, I'd like to ship early next week 🙇

File: client/my-sites/plans-features-main/index.jsx [hunk 1]
```diff
@@ -36,8 +37,27 @@ import { plansLink } from 'lib/plans';
...
+		 * Happychat does not update with the selected site right now :(
+		 * This ensures that Happychat groups are correct in case we switch sites while on the plans
+		 * page, for example between a Jetpack and Simple site.
+		 *
+		 * @TODO: When happychat correctly handles site switching, remove selectSiteId action.
+		 */
+		const siteId = get( this.props, [ 'site', 'ID' ] );
+		const nextSiteId = get( nextProps, [ 'site', 'ID' ] );
+		if ( siteId !== nextSiteId && nextSiteId ) {
+			this.props.selectSiteId( nextSiteId );
```
Comment (lamosty) [inline client/my-sites/plans-features-main/index.jsx]: I'm not a big fan of the name `selectSiteId`. From a quick glance, it looks like it's an action for changing the selected site ID globally. But looking deeper, it's just for changing the selected site for Happychat. Could we maybe import it as something more descriptive, like `selectSiteIdForHappychat`?

Comment (sirreal) [reply client/my-sites/plans-features-main/index.jsx]: I agree, I'll swap for a clearer alternative.

**Comment by sirreal** [comment]
e2e failures appear to be related to buggy base.

Rebased, when e2e passes 🤞 I'll merge

**Pull request merged**